### PR TITLE
DEV-2979: Revise frequency editor

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Revises FrequencyEditor to match new Figma designs and migrate to TypeScript.

#### Why are we doing this? How does it help us?

Improved UX, continues TS migration.

#### How should this be manually tested? Please include detailed step-by-step instructions.

1. Edit a contribution page and its frequency element.
2. Disable a frequency using a toggle on the left and save changes.
3. Confirm the frequency no longer shows up as an option on the published page.
4. Confirm making a one-time contribution and recurring contribution work properly (e.g. the correct frequency is recorded).
5. Edit the page and frequency element.
6. Toggle back on the frequency you disabled in step 2 and save changes.
7. Confirm that the frequency is back on the published page.
8. Edit the page and frequency amount.
9. Change the default frequency using the radio buttons on the right and save changes.
10. Confirm the published page defaults to the correct contribution frequency.
11. Edit the page and frequency amount.
12. Disable all contribution frequencies.
13. Confirm that the Update button is disabled and an error message appears.
14. Toggle on a frequency and confirm you're able to save changes again.
15. Edit the page and frequency amount.
16. Toggle off the frequency currently selected as default.
17. Confirm that the editor selects the first frequency available as default.
18. Save changes and confirm that the default frequency on the published page matches what the editor showed.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

Yes. We might need new screenshots but the wording on controls has only changed in small ways.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

See above re: screenshots.

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2979

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

File: spa/cypress/e2e/02-donationPageEdit.cy.js [hunk 1]
```diff
@@ -182,7 +182,7 @@ describe('Contribution page edit', () => {
       cy.getByTestId('element-properties').should('exist');
     });
 
-    describe('Frequency editor', () => {
+    describe.only('Frequency editor', () => {
```
Comment (GuiMend) [inline spa/cypress/e2e/02-donationPageEdit.cy.js:185]: Forgot to remove the `only`
```suggestion
    describe('Frequency editor', () => {
```

File: spa/src/components/base/Radio/Radio.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Radio as MuiRadio, RadioProps as MuiRadioProps } from '@material-ui/core';
+import styled from 'styled-components';
+// See https://create-react-app.dev/docs/adding-images-fonts-and-files/#adding-svgs
+import { ReactComponent as CheckedIcon } from './checked.svg';
+import { ReactComponent as UncheckedIcon } from './unchecked.svg';
```
Comment (GuiMend) [inline spa/src/components/base/Radio/Radio.tsx:5]: Don't we have these icons in one of the icon libs we are using?

File: spa/src/components/pageEditor/elementEditors/frequency/FrequencyEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+              <Switch checked={enabled.includes('month')} onChange={(event) => handleEnabledChange('month', event)} />
+            }
+            label="Monthly payments enabled"
+          />
+          <ToggleFormControlLabel
+            control={
+              <Switch checked={enabled.includes('year')} onChange={(event) => handleEnabledChange('year', event)} />
+            }
+            label="Yearly payments enabled"
+          />
```
Comment (GuiMend) [inline spa/src/components/pageEditor/elementEditors/frequency/FrequencyEditor.tsx:95]: No need to change, but could create an array with the different values and map it here as the component structure is the same for all 3.

File: spa/cypress/e2e/02-donationPageEdit.cy.js (hunk 1 above)
Comment (nrh-cklimas) [reply spa/cypress/e2e/02-donationPageEdit.cy.js]: Yep, faceplanted on this one. Fixed in 5c868b0c99f866bfd5758299f2286ff16a187648

File: spa/src/components/pageEditor/elementEditors/frequency/FrequencyEditor.test.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        onChangeElementContent,
+        elementContent: [unrelatedContent, frequencyContent]
+      });
+      expect(onChangeElementContent).not.toBeCalled();
+      userEvent.click(screen.getByRole('checkbox', { name }));
+      expect(onChangeElementContent.mock.calls).toEqual([[[unrelatedContent]]]);
+    });
+  });
+
+  describe.each([['one_time', 'One time', { value: CONTRIBUTION_INTERVALS.ONE_TIME, displayName: 'One time' }]])(
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/pageEditor/elementEditors/frequency/FrequencyEditor.test.tsx]: Fixed in 5c868b0c99f866bfd5758299f2286ff16a187648

File: spa/src/components/pageEditor/elementEditors/frequency/FrequencyEditor.tsx (hunk 3 above)
Comment (nrh-cklimas) [reply spa/src/components/pageEditor/elementEditors/frequency/FrequencyEditor.tsx:95]: Yeah, thought about this but it felt borderline with 3 inputs.

**Review by GuiMend — APPROVED** [review]

File: spa/src/components/base/Radio/Radio.tsx (hunk 2 above)
Comment (nrh-cklimas) [reply spa/src/components/base/Radio/Radio.tsx]: :+1: The checked appearance is basically the inverse of the Material icon.

**Review by x110dc — APPROVED** [review]

**Pull request merged**